nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -587,27 +601,166 @@ in {
 
     services.nginx = lib.mkIf cfg.configureNginx {
       enable = true;
-      recommendedProxySettings = true; # required for redirections to work
+
+      commonHttpConfig = ''
+        proxy_cache_path /var/cache/nginx levels=1:2 keys_zone=all_cache:10m inactive=7d max_size=1g;
+      '';
+
       virtualHosts."${cfg.localDomain}" = {
         root = "${cfg.package}/public/";
         forceSSL = true; # mastodon only supports https
         enableACME = true;
 
-        locations."/system/".alias = "/var/lib/mastodon/public-system/";
-
         locations."/" = {
           tryFiles = "$uri @proxy";
+          priority = 1100;
         };
 
-        locations."@proxy" = {
-          proxyPass = (if cfg.enableUnixSocket then "http://unix:/run/mastodon-web/web.socket" else "http://127.0.0.1:${toString(cfg.webPort)}");
-          proxyWebsockets = true;
+        locations."= /sw.js" = {
+          tryFiles = "$uri =404";
+          priority = 2110;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=604800, must-revalidate';
+            ${nginxCommonHeaders}
+          '';
+        };
+
+        locations."= /sw.js.map" = {
```

Comment:
The upstream config doesn't define `/sw.js.map`. Why's it needed?

Reply:
This `locations` added in this PR - https://github.com/mastodon/mastodon/pull/19644

Reply:
That PR hasn't been accepted by upstream yet. It's also not obvious to me whether setting the cache headers for the sourcemap is beneficial, since I don't expect it to be downloaded during normal use of the app ...?

Reply:
This file was added only in version 4, acts on the same principle as the sw.js file.
Developers have not yet had time to consider PR, probably has a low priority.

Reply:
Then it would be better to leave it out of this PR here, to keep the change as small as possible.

Reply:
I think it's better to make necessary changes right away. Sooner or later will make these changes in upstream.

Reply:
Is one change (updating the config according to https://github.com/mastodon/mastodon/pull/19438) impossible without https://github.com/mastodon/mastodon/pull/19644? If not, let's leave the changes separate. All this is already complicated enough. See, we have to arrive at a solution that not only you understand, but that all other mastodon maintainers also understand and can work with.

Reply:
Removing this location.